cluster-autoscaler/core/scaledown/unneeded/nodes_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unneeded
```

Comment:
I think a unittest for RemovableAt / unremovableReason would be a nice bonus. The logic in unremovableReason seems much more testable now. I realize this PR is just moving existing code around, so I'm not going to block on this, but if you decide to make more changes in this PR maybe that could be an extra target of opportunity.